app/controllers/forumthreads_controller.rb

```diff
@@ -80,6 +85,20 @@ def destroy
     redirect_to @thread.forum
   end
 
+  def search
+  end
+
+  def search_redirect
```

Comment:
Again, URL parameters aren't meant to be user friendly or good-looking. It's perfectly fine to have parameters with empty values in there.

> If someone wanted to share a link, it would take them some time to get the unneeded params out.

It wouldn't take them any time at all because they'd just share it like that, nobody cares about URLs. Trust me, I get image links [like this one](https://images.google.de/imgres?imgurl=https%3A%2F%2Fi.ytimg.com%2Fvi%2FVXEtJLuxjks%2Fmaxresdefault.jpg&imgrefurl=https%3A%2F%2Fwww.youtube.com%2Fwatch%3Fv%3DVXEtJLuxjks&docid=Iz53a8I8AuiVgM&tbnid=6azaDVw_digYtM%3A&vet=1&w=1920&h=1080&source=sh%2Fx%2Fim) all the time.

You can use `[...].drop_while(&:blank?).any?`.